### PR TITLE
Add object events to supporter, through nonprofit

### PR DESCRIPTION
**Pull request by awadyas**
**NOTE: DO NOT discuss internal CommitChange information in your PR; this PR will be public.
Link back to the issue in the Tix repo when you need to do that.**

For https://github.com/CommitChange/tix/issues/3941

### CODE REVIEW

File: app/models/supporter.rb [hunk 1]
```diff
@@ -66,6 +66,7 @@ class Supporter < ActiveRecord::Base
   has_many :activities, dependent: :destroy
   has_many :tickets
   has_many :recurring_donations
+  has_many :object_events, through: :nonprofit, source: :associated_object_events
```
Comment (awadyas) [inline app/models/supporter.rb]: @wwahammy not sure if this is the best way of doing this. LMK what you think

Comment (wwahammy) [inline app/models/supporter.rb]: No, it should be connected to object_events like so:
I'd recommend following how it's done in Payout:

```suggestion
  has_many :object_events, as: :event_entity
```

**Review by wwahammy — CHANGES_REQUESTED** [review]

File: spec/models/supporter_spec.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
   it { is_expected.to have_many(:addresses).class_name("SupporterAddress")}
   it { is_expected.to belong_to(:primary_address).class_name("SupporterAddress")}
+  it { is_expected.to have_many(:object_events).through(:nonprofit) }
```
Comment (awadyas) [inline spec/models/supporter_spec.rb]:
```suggestion
  it { is_expected.to have_many(:object_events).as(:event_entity) }
```

**Review by wwahammy — APPROVED** [review]
LGTM

**Pull request merged**